autometa/taxonomy/gtdb.py

```diff
@@ -0,0 +1,362 @@
+"""
+# License: GNU Affero General Public License v3 or later
+# A copy of GNU AGPL v3 should have been included in this software package in LICENSE.txt.
+
+File containing definition of the GTDB class and containing functions useful for handling GTDB taxonomy databases
+"""
+
+import gzip
+import logging
+import os
+import re
+import subprocess
+import shutil
+import tarfile
+
+from typing import Dict, List, Set, Tuple
+from itertools import chain
+from tqdm import tqdm
+
+import pandas as pd
+import multiprocessing as mp
+
+from autometa.common.utilities import file_length
+from autometa.common.external import diamond
+from autometa.taxonomy.database import TaxonomyDatabase
+
+
+logger = logging.getLogger(__name__)
+
+
+def is_gz_file(filepath) -> bool:
+    """
+    Check if the given file is gzipped compressed or not.
+
+    Parameters
+    ----------
+    filepath : str
+        Filepath to check
+
+    Returns
+    -------
+    bool
+        True if file is gzipped else False
+    """
+    # https://stackoverflow.com/a/47080739
+    with open(filepath, "rb") as test_f:
+        return test_f.read(2) == b"\x1f\x8b"
+
+
+def create_gtdb_db(reps_faa: str, dbdir: str):
+
+    if reps_faa.endswith(".tar.gz"):
+        logger.debug(
+            f"Extracting tarball containing GTDB ref genome animo acid data sequences to: {dbdir}/protein_faa_reps"
+        )
+        tar = tarfile.open(reps_faa)
+        tar.extractall(path=dbdir)
+        tar.close()
+        logger.debug("Extraction done.")
+        reps_faa = dbdir
+
+    faa_index = {}
+    for dirpath, dirs, files in os.walk(reps_faa):
+        for file in files:
+            if file.endswith("_protein.faa"):
+                faa_file = os.path.join(dirpath, file)
+                # Regex to get the genome accession from the file path
+                acc = re.search(r"GCA_\d+.\d?|GCF_\d+.\d?", file)
+                faa_index[faa_file] = acc.group()
+
+    # Create dbdir if it doesn't exist
+    if not os.path.isdir(dbdir):
+        os.makedirs(dbdir)
+
+    logger.debug("Merging faa files.")
```

Comment:
More context could be provided here.

```suggestion
    logger.debug(f"Merging {len(faa_index):,} faa files.")
```